session/session.go

```diff
@@ -0,0 +1,144 @@
+// If the BITRISE_PORTAL_DATA_JSON is provided (for debug purposes) it will use that instead.
+func GetSession() (string, error) {
+	portalData, err := getDeveloperPortalData(os.Getenv("BITRISE_BUILD_URL"), os.Getenv("BITRISE_BUILD_API_TOKEN"))
+	if err != nil {
+		return "", err
+	}
+
+	cookies := convertDesCookie(portalData.SessionCookies["https://idmsa.apple.com"])
+	session := strings.Join(cookies, "")
+	return session, nil
+}
+
+func getDeveloperPortalData(buildURL, buildAPIToken string) (PortalData, error) {
+	var developerPortalData PortalData
+
+	portalDataJSON, exists := os.LookupEnv("BITRISE_PORTAL_DATA_JSON")
+	if exists && portalDataJSON != "" {
+		return developerPortalData, json.Unmarshal([]byte(portalDataJSON), &developerPortalData)
+	}
+
+	if buildURL == "" {
+		return PortalData{}, fmt.Errorf("BITRISE_BUILD_URL env is not exported")
+	}
+
+	if buildAPIToken == "" {
+		return PortalData{}, fmt.Errorf("BITRISE_BUILD_API_TOKEN env is not exported")
+	}
+
+	req, err := http.NewRequest(http.MethodGet, fmt.Sprintf("%s/apple_developer_portal_data.json", buildURL), nil)
+	if err != nil {
+		return PortalData{}, err
+	}
+
+	req.Header.Add("BUILD_API_TOKEN", buildAPIToken)
+
+	if _, err := performRequest(req, &developerPortalData); err != nil {
+		return PortalData{}, fmt.Errorf("Falied to fetch portal data from Bitrise, error: %s", err)
+	}
+	return developerPortalData, nil
+}
+
+func convertDesCookie(cookies []Cookie) []string {
+	var convertedCookies []string
+	for _, c := range cookies {
+		if convertedCookies == nil {
+			convertedCookies = append(convertedCookies, "---"+"\n")
+		}
+
+		if c.ForDomain == nil {
+			b := true
+			c.ForDomain = &b
+		}
+
+		convertedCookie := strings.Replace(cookieTemplate, "<NAME>", c.Name, 1)
+		convertedCookie = strings.Replace(convertedCookie, "<VALUE>", c.Value, 1)
+		convertedCookie = strings.Replace(convertedCookie, "<DOMAIN>", c.Domain, 1)
+		convertedCookie = strings.Replace(convertedCookie, "<FOR_DOMAIN>", strconv.FormatBool(*c.ForDomain), 1)
+		convertedCookie = strings.Replace(convertedCookie, "<PATH>", c.Path, 1)
+
+		convertedCookies = append(convertedCookies, convertedCookie+"\n")
+	}
+
+	return convertedCookies
+}
+
+func performRequest(req *http.Request, requestResponse interface{}) ([]byte, error) {
+	client := http.Client{}
+	response, err := client.Do(req)
+	if err != nil {
+		// On error, any Response can be ignored
+		return nil, fmt.Errorf("failed to perform request, error: %s", err)
+	}
+
+	// The client must close the response body when finished with it
+	defer func() {
+		if cerr := response.Body.Close(); cerr != nil {
+			log.Warnf("Failed to close response body, error: %s", cerr)
+		}
+	}()
+
+	body, err := ioutil.ReadAll(response.Body)
+	if err != nil {
+		return nil, fmt.Errorf("failed to read response body, error: %s", err)
+	}
+
+	if response.StatusCode < http.StatusOK || response.StatusCode > http.StatusMultipleChoices {
```

Comment:
Think you meant response.StatusCode >= http.StatusMultipleChoice

Reply:
Please find out (this is our server) what status code will be sent back for successful actions by the server and accept only that.

Reply:
Fixed in:
https://github.com/bitrise-io/steps-deploy-to-itunesconnect-deliver/pull/61/commits/7da4ffe59c06b9160bbafa2ca0510409663c6d64